khmer/_oxli/graphs.pyx

```diff
@@ -1,20 +1,194 @@
+
+    def get_median_count(self, sequence):
+        """median, average, and stddev of the k-mer counts in sequence."""
+        data = self._valid_sequence(sequence)
+        cdef BoundedCounterType med = 0
+        cdef float average = 0
+        cdef float stddev = 0
+
+        deref(self.c_table).get_median_count(data, med, average, stddev)
+        return (med, average, stddev)
+
+    def get_kmer_hashes(self, sequence):
+        """Retrieve hashes of all k-mers in sequence.
+
+        Hashes are returned in the same order as k-mers appear in sequence.
+        """
+        data = self._valid_sequence(sequence)
+        cdef vector[HashIntoType] hashes
+        deref(self.c_table).get_kmer_hashes(data, hashes)
+        return hashes
+
+    def trim_on_abundance(self, sequence, abundance):
+        """"Trim sequence at first k-mer below the given abundance."""
+        data = self._valid_sequence(sequence)
+        trimmed_at = deref(self.c_table).trim_on_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def trim_below_abundance(self, sequence, abundance):
+        """"Trim sequence at first k-mer above the given abundance."""
+        data = self._valid_sequence(sequence)
+        trimmed_at = deref(self.c_table).trim_below_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def find_spectral_error_positions(self, sequence, max_count):
+        """"Identify positions of low-abundance k-mers."""
+        data = self._valid_sequence(sequence)
+        posns = (deref(self.c_table).find_spectral_error_positions(data,
+                                                                   max_count))
+        return posns
+
+    def consume_seqfile_with_reads_parser(self, FastxParser read_parser):
+        """Count all k-mers from read_parser."""
+        cdef unsigned long long n_consumed = 0
+        cdef unsigned int total_reads = 0
+
+        deref(self.c_table).consume_seqfile[CpFastxReader](read_parser._this,
+                                                           total_reads,
+                                                           n_consumed)
+
+    def consume_seqfile(self, file_name):
+        """Count all k-mers from file_name."""
+        cdef unsigned long long n_consumed = 0
+        cdef unsigned int total_reads = 0
+
+        read_parser = FastxParser(file_name)
+        deref(self.c_table).consume_seqfile[CpFastxReader](read_parser._this,
+                                                           total_reads,
+                                                           n_consumed)
+
+    def abundance_distribution(self, file_name, tracking):
+        """Calculate the k-mer abundance distribution of the given file_name."""
+        read_parser = FastxParser(file_name)
+        cdef uint64_t * x = deref(self.c_table).abundance_distribution[CpFastxReader](
+                read_parser._this, (<CPyHashtable_Object>tracking).hashtable)
```

Comment:
Need help again @camillescott :-/ `tracking` is a `Nodegraph` which I'd like to somehow unpack so I can take the `hashtable` attribute and pass it along.

A bit like:
```c++
khmer_KHashtable_Object * tracking_obj = NULL;
PyArg_ParseTuple(args, "O!", &khmer_KHashtable_Type, &tracking_obj)
// now use tracking_obj->hashtable
```

but how? It seems silly that we can't reach into these extension types 😕